src/progress_bar_exercise/ProgressBar/progress-bar.component.js

```diff
@@ -0,0 +1,41 @@
+import { useCss } from "kremling";
+import React, { useEffect, useState } from "react";
+
+export default function ProgressBar({ loading }) {
+  const [seconds, setSeconds] = useState(0)
+  const [opacity, setOpacity] = useState(1)
+
+  useEffect(() => {
+    if (seconds && !loading) { // loading was toggled to false and we have a count going
+      setSeconds(16)
+      const timer = setTimeout(() => setOpacity(0), 3000)
+      return () => clearTimeout(timer)
+    } else if (loading) { // loading is true
+      if (seconds > 15 || seconds == 0) setSeconds(1)
+      setOpacity(1)
+      const timer = setInterval(() => {
+        if (seconds < 15) setSeconds(seconds + 1)
+      }, 1000)
+      return () => clearInterval(timer)
```

Comment:
The second condition describes the start of the 15 second crawl of the progress bar. It supports initial load and reclick (hence seconds > 15)